rtengine/dcraw.cc

```diff
@@ -6549,9 +6549,7 @@ int CLASS parse_tiff_ifd (int base)
 	tiff_ifd[ifd].tile_length = getint(type);
 	break;
       case 324:				/* TileOffsets */
-	tiff_ifd[ifd].offset = len > 1 ? ftell(ifp) : get4();
-	if (len == 1)
-	  tiff_ifd[ifd].tile_width = tiff_ifd[ifd].tile_length = 0;
+	tiff_ifd[ifd].offset = len > 1 ? ftell(ifp) : getint(type);
```

Comment:
Seems like this needs to be changed to:
```suggestion
	tiff_ifd[ifd].offset = ftell(ifp);
```
I'll need to read the TIFF specification again...

Reply:
...no, this is wrong...